compiler/rustc_lint/src/builtin.rs

```diff
@@ -1814,7 +1814,7 @@ declare_lint! {
     "detects edition keywords being used as an identifier",
     @future_incompatible = FutureIncompatibleInfo {
         reason: FutureIncompatibilityReason::EditionError(Edition::Edition2024),
-        reference: "issue #49716 <https://github.com/rust-lang/rust/issues/49716>",
+        reference: "<https://doc.rust-lang.org/nightly/edition-guide/rust-2024/gen-keyword.html>",
```

Comment:
Question (in general): we would have to change this to a non-nightly path once 2024 edition guide is stabilized right?

Reply:
Not really, no. We use `/nightly/` URLs all over the compiler. The edition guide in particular is somewhat stable, in that it is unlikely `/nightly/` will introduce something that is *wrong* for a stable user. Edition changes are typically (though not always) fixed once it launches. There's also the benefit that users will see the most up-to-date docs with any fixes or clarifications.

That being said, I would think that `/stable/` URLs would also be fine once on the stable channel, but the compiler doesn't really have that capability. In cargo, we use an [automated linking system](https://github.com/rust-lang/cargo/blob/addcc8ca715bc7fe20df66afd6efbf3c77ef43f8/src/cargo/core/features.rs#L1405-L1414) to choose the channel. But I don't think that would work very well with compiletest validating the output.

Reply:
Right, that is fine, thanks for the context!